model/vfs/vfsswift/impl_v3.go

```diff
@@ -591,6 +594,101 @@ func (sfs *swiftVFSV3) RevertFileVersion(doc *vfs.FileDoc, version *vfs.Version)
 	return sfs.Indexer.DeleteVersion(version)
 }
 
+func (sfs *swiftVFSV3) CopyFileFromOtherFS(
```

Comment:
Looks like a great part of this function could be shared with `CreateFile` (as a guard method maybe?).

Reply:
Seems also that `CopyFile` could benefit from some of these checks.

Reply:
Looks like the end of the function could also be shared (versions management, modifications when uploading to the trash…).

Reply:
The exception is maybe the version management, which is a bigger block, I will look for it.

Reply:
I probably overlooked things but I wanted to bring it up to you. Great if this can lead to some code sharing :).